src/PrestaShopBundle/Resources/config/services/bundle/form/form_type.yml

```diff
@@ -227,7 +227,8 @@ services:
 
   PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType:
     arguments:
-      $stringArrayToIntegerArrayDataTransformer: '@PrestaShopBundle\Form\DataTransformer\DefaultLanguageToFilledArrayDataTransformer'
+      $defaultLanguageToFilledArrayDataTransformer: '@PrestaShopBundle\Form\DataTransformer\DefaultLanguageToFilledArrayDataTransformer'
+      $stringArrayToIntegerArrayDataTransformer: '@PrestaShopBundle\Form\DataTransformer\StringArrayToIntegerArrayDataTransformer'
```

Comment:
Wow great catch @Codencode 
I searched for the PR that modified this and found it here https://github.com/PrestaShop/PrestaShop/pull/31391/files#diff-ce2d57a2ac02f46a9cd2eea65b89142c51e7df0a68afc4f9d1e9e01a53999f75L351

So actually your PR should be even simpler, it's just that the wrong transformer was assigned during the refacto, the ShopChoiceTreeType has nothing to do with multi lang values so `DefaultLanguageToFilledArrayDataTransformer` shouldn't be passed at all

```suggestion
      $stringArrayToIntegerArrayDataTransformer: '@PrestaShopBundle\Form\DataTransformer\StringArrayToIntegerArrayDataTransformer'
```

Reply:
@jolelievre,
sorry, the issue was due to the fact that I "obviously" need to remove the instructions related to the `defaultLanguageToFilledArrayDataTransformer` attribute in the `PrestaShopBundle\Form\Admin\Type\ShopChoiceTreeType` class.